### ISSUE TITLE
Datasets and their associated rules disappearing

### ISSUE THREAD

**Issue by StevenCTimm** [DUNE/data-mgmt-ops#483]
Over the weekend we created all these datasets

[dunepro@dunesl7gpvm01 mcruciosam]$ cat istapedatasets
fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_limit-1_
fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip5000_limit5000_1572
fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_end_1574
fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_end_1574
fardet-hd:fardet-hd-reco2ana_ritm1780305_nue_fhc_skip15000_end_1575
fardet-hd:fardet-hd-reco2_ritm1780305_nue_fhc_skip15000_end_1575
fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip5000_limit5000_1572
fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_limit-1_1573
fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_limit-1_1573
fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip15000_end_1576
fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip15000_end_1576
fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_limit-1_
fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip10000_limit5000_1577
fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip10000_limit5000_1577

They all were initially created with one-day rules which I then extended to 7 days
55b53a0c52704f109124a45854bf5022  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_limit-1_                                                                                     OK[0/0/0]                  istape=False\decommissioned=True  1         N/A         2024-02-25 15:27:18  2024-02-18 01:45:57
cc4e55741ff1424983776ae596ac597d  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip5000_limit5000_1572                                                                            REPLICATING[4400/597/0]    istape=False\decommissioned=True  1         N/A         2024-02-25 20:53:01  2024-02-18 18:24:54
fccd9fc49f4748dbb1b7f72134a29fd3  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_end_1574                                                                                  REPLICATING[4020/829/0]    istape=False\decommissioned=True  1         N/A         2024-02-25 20:54:00  2024-02-18 18:53:58
235a8b9982ae4efdbeeac43c3c75a449  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_end_1574                                                                                     REPLICATING[4008/841/0]    istape=False\decommissioned=True  1         N/A         2024-02-25 20:54:32  2024-02-18 18:54:00
cfb821fb3c4340dfb29bba609a50e51d  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_nue_fhc_skip15000_end_1575                                                                                  REPLICATING[4692/1132/0]   istape=False\decommissioned=True  1         N/A         2024-02-25 20:54:59  2024-02-18 18:58:39
1afde48486d5423c8e11c675f757d49a  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_nue_fhc_skip15000_end_1575                                                                                     REPLICATING[4667/1157/0]   istape=False\decommissioned=True  1         N/A         2024-02-25 20:55:20  2024-02-18 18:58:35
07dbd00272724f389b41d58153c8f702  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip5000_limit5000_1572                                                                               REPLICATING[4400/597/0]    istape=False\decommissioned=True  1         N/A         2024-02-25 21:00:07  2024-02-18 18:24:51
ac9e4f003c584e90be1ff835402ae649  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anu_rhc_skip15000_limit-1_1573                                                                                 OK[0/0/0]                  istape=False\decommissioned=True  1         N/A         2024-02-25 21:00:07  2024-02-18 18:39:33
4c1e2a00e61a419687777d2f465561c7  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_limit-1_1573                                                                              OK[0/0/0]                  istape=False\decommissioned=True  1         N/A         2024-02-25 21:00:08  2024-02-18 18:39:37
6d9f5c80690546dd91742155a5331070  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip15000_end_1576                                                                                    REPLICATING[3272/1313/0]   istape=False\decommissioned=True  1         N/A         2024-02-25 22:52:39  2024-02-18 21:47:27
66eab4b76f954a0f8dcea4ecf4d52b8e  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip15000_end_1576                                                                                 REPLICATING[3279/1306/0]   istape=False\decommissioned=True  1         N/A         2024-02-25 22:53:19  2024-02-18 21:47:31
6025480e2e424635b64f291f97306fb4  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anu_rhc_skip15000_limit-1_                                                                                  OK[0/0/0]                  istape=False\decommissioned=True  1         N/A         2024-02-25 22:54:00  2024-02-18 01:45:55
dc48b976a5f740118b35cca8403a1f56  dunepro    fardet-hd:fardet-hd-reco2_ritm1780305_anue_rhc_skip10000_limit5000_1577                                                                              REPLICATING[2962/2036/0]   istape=False\decommissioned=True  1         N/A         2024-02-25 22:54:29  2024-02-18 21:48:59
062126d9e87b4c129b5afe4a1d5c45d8  dunepro    fardet-hd:fardet-hd-reco2ana_ritm1780305_anue_rhc_skip10000_limit5000_1577                                                                           REPLICATING[2991/2007/0]   istape=False\decommissioned=True  1         N/A         2024-02-26 16:48:58  2024-02-18 21:48:57
@FNAL.GOV

But even though the rules were lengthened most of them (all but 4) have gone away and the dataset with them.

This happened once before under the previous rucio server on feb 4-5

**Comment by StevenCTimm** [comment]
And it's now also the case from looking at the logs that some of the files in those datasets were deleted too.
We have stopped greedyDeletion to save the rest.

**Comment by StevenCTimm** [comment]
#482 is related to this.  Doug found a remnant of a file where the data set had been deleted but the file was still there.

Next step is to give a reproducer to the rucio mattermost people and then have James also look at it.
this is a bug that's got to be fixed..without this then we don't know if any expiration date is real or not.

**Comment by StevenCTimm** [comment]
here's a reproducer:

[dunepro@dunesl7gpvm01 mcruciosam]$ rucio add-dataset test:test_expire_rules
Added test:test_expire_rules
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio -v -a test upload --rse DUNE_US_FNAL_DISK_STAGE --lifetime 172000 --scope test --register-after-upload --protocol davs --name 1gbtestfile.20240219.bb /tmp/1gbtestfile.20240219.bb 
2024-02-19 21:20:31,964	DEBUG	baseclient.py	No trace_host passed. Using rucio_host instead
2024-02-19 21:20:31,964	DEBUG	baseclient.py	No creds passed. Trying to get it from the config file.
2024-02-19 21:20:31,964	DEBUG	baseclient.py	HTTPS is required, but no ca_cert was passed. Trying to get it from X509_CERT_DIR.
2024-02-19 21:20:31,964	DEBUG	baseclient.py	HTTPS is required, but no ca_cert was passed and X509_CERT_DIR is not defined. Trying to get it from the config file.
2024-02-19 21:20:31,965	DEBUG	baseclient.py	No VO passed. Trying to get it from environment variable RUCIO_VO.
2024-02-19 21:20:31,965	DEBUG	baseclient.py	No VO found. Trying to get it from the config file.
2024-02-19 21:20:31,965	DEBUG	baseclient.py	No VO found. Using default VO.
2024-02-19 21:20:31,965	DEBUG	baseclient.py	got token from file
2024-02-19 21:20:34,569	DEBUG	uploadclient.py	Num. of files that upload client is processing: 1
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP request: GET https://dune-rucio.fnal.gov/rses/?expression=DUNE_US_FNAL_DISK_STAGE
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:34,570	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:34,683	DEBUG	baseclient.py	HTTP Response: 200 OK
2024-02-19 21:20:34,746	DEBUG	uploadclient.py	Input validation done.
2024-02-19 21:20:34,746	INFO	Preparing upload for file 1gbtestfile.20240219.bb
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP request: GET https://dune-rucio.fnal.gov/rses/DUNE_US_FNAL_DISK_STAGE/attr/
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:34,746	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:34,764	DEBUG	baseclient.py	HTTP Response: 200 OK
2024-02-19 21:20:34,764	DEBUG	uploadclient.py	wan domain is used for the upload
2024-02-19 21:20:34,909	DEBUG	logging.py	gfal.Default: connecting to storage
2024-02-19 21:20:35,228	DEBUG	logging.py	gfal.Default: checking if file exists None
2024-02-19 21:20:35,229	DEBUG	rsemanager.py	Checking if davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb exists
2024-02-19 21:20:35,229	DEBUG	logging.py	gfal.Default: checking if file exists davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb
2024-02-19 21:20:35,400	DEBUG	logging.py	gfal.Default: closing protocol connection
2024-02-19 21:20:35,401	DEBUG	uploadclient.py	[{'hostname': 'fndca1.fnal.gov', 'scheme': 'davs', 'port': 2880, 'prefix': '/dune/persistent/staging', 'impl': 'rucio.rse.protocols.gfal.Default', 'domains': {'lan': {'read': 2, 'write': 1, 'delete': 1}, 'wan': {'read': 2, 'write': 1, 'delete': 1, 'third_party_copy_read': 1, 'third_party_copy_write': 1}}, 'extended_attributes': None}]
2024-02-19 21:20:35,401	INFO	Trying upload with davs to DUNE_US_FNAL_DISK_STAGE
2024-02-19 21:20:35,401	DEBUG	uploadclient.py	Processing upload with the domain: wan
2024-02-19 21:20:35,401	DEBUG	logging.py	gfal.Default: connecting to storage
2024-02-19 21:20:35,403	DEBUG	uploadclient.py	The PFN created from the LFN: davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb
2024-02-19 21:20:35,403	DEBUG	logging.py	gfal.Default: checking if file exists davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb
2024-02-19 21:20:35,495	DEBUG	logging.py	gfal.Default: checking if file exists davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload
2024-02-19 21:20:35,533	DEBUG	utils.py	put: Attempt 1
2024-02-19 21:20:35,533	DEBUG	logging.py	gfal.Default: uploading file from 1gbtestfile.20240219.bb to davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload
 error in request of checking file QoS:   is not a valid HTTP or Webdav URL

(process:13568): GLib-WARNING **: 21:20:35.534: GError set over the top of a previous GError or uninitialized memory.
This indicates a bug in someone's code. You must ensure an error is NULL before it's set.
The overwriting error message was:  \x01 is not a valid HTTP or Webdav URL
2024-02-19 21:20:41,158	INFO	Successful upload of temporary file. davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload
2024-02-19 21:20:41,158	DEBUG	uploadclient.py	skip_upload_stat=False
2024-02-19 21:20:41,158	DEBUG	uploadclient.py	stat: pfn=davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload
2024-02-19 21:20:41,158	DEBUG	logging.py	gfal.Default: getting stats of file davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload
2024-02-19 21:20:41,266	DEBUG	uploadclient.py	Filesize: Expected=1024000000 Found=1024000000
2024-02-19 21:20:41,266	DEBUG	uploadclient.py	Checksum: Expected=93b40001 Found=93b40001
2024-02-19 21:20:41,266	DEBUG	uploadclient.py	Renaming file davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload to davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb
2024-02-19 21:20:41,266	DEBUG	logging.py	gfal.Default: renaming file from davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb.rucio.upload to davs://fndca1.fnal.gov:2880/dune/persistent/staging/test/c7/f5/1gbtestfile.20240219.bb
2024-02-19 21:20:41,355	DEBUG	logging.py	gfal.Default: closing protocol connection
2024-02-19 21:20:41,356	DEBUG	uploadclient.py	Upload done.
2024-02-19 21:20:41,356	INFO	Successfully uploaded file 1gbtestfile.20240219.bb
/cvmfs/dune.opensciencegrid.org/products/dune/python_requests/v2_25_0/NULL/lib/python3/site-packages/urllib3/connectionpool.py:1045: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dune-rucio.fnal.gov'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
2024-02-19 21:20:41,402	DEBUG	uploadclient.py	Registering file
2024-02-19 21:20:41,402	DEBUG	baseclient.py	HTTP request: GET https://dune-rucio.fnal.gov/accounts/test/scopes/
2024-02-19 21:20:41,402	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:41,402	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:41,403	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:41,403	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:41,403	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:41,403	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:41,494	DEBUG	baseclient.py	HTTP Response: 200 OK
2024-02-19 21:20:41,495	DEBUG	uploadclient.py	Skipping dataset registration
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP request: GET https://dune-rucio.fnal.gov/dids/test/1gbtestfile.20240219.bb/meta
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:41,495	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:41,571	DEBUG	baseclient.py	HTTP Response: 404 NOT FOUND
2024-02-19 21:20:41,572	DEBUG	baseclient.py	Response text (length=124): [{"ExceptionClass": "DataIdentifierNotFound", "ExceptionMessage": "Data identifier 'test:1gbtestfile.20240219.bb' not found"}]
2024-02-19 21:20:41,572	DEBUG	uploadclient.py	File DID does not exist
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP request: POST https://dune-rucio.fnal.gov/replicas
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:41,572	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:41,572	DEBUG	baseclient.py	Request data (length=283): [{"rse": "DUNE_US_FNAL_DISK_STAGE", "files": [{"scope": "test", "name": "1gbtestfile.20240219.bb", "bytes": 1024000000 ... ty": true}]
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP Response: 201 CREATED
2024-02-19 21:20:41,731	INFO	Successfully added replica in Rucio catalogue at DUNE_US_FNAL_DISK_STAGE
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP request: POST https://dune-rucio.fnal.gov/rules/
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  X-Rucio-Auth-Token: [hidden]
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  X-Rucio-VO: def
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  Connection: Keep-Alive
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  User-Agent: rucio-clients/33.3.0
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  X-Rucio-Script: rucio::-v
2024-02-19 21:20:41,731	DEBUG	baseclient.py	HTTP header:  X-Rucio-Account: test
2024-02-19 21:20:41,731	DEBUG	baseclient.py	Request data (length=455): [{"dids": [{"scope": "test", "name": "1gbtestfile.20240219.bb"}], "copies": 1, "rse_expression": "DUNE_US_FNAL_DISK_ST ... ta": null}]
2024-02-19 21:20:42,012	DEBUG	baseclient.py	HTTP Response: 201 CREATED
2024-02-19 21:20:42,012	INFO	Successfully added replication rule at DUNE_US_FNAL_DISK_STAGE
Completed in 10.0489 sec.
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:test_expire_rules
ID    ACCOUNT    SCOPE:NAME    STATE[OK/REPL/STUCK]    RSE_EXPRESSION    COPIES    SIZE    EXPIRES (UTC)    CREATED (UTC)
----  ---------  ------------  ----------------------  ----------------  --------  ------  ---------------  ---------------
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:1gbtestfile.20240219.bb
ID                                ACCOUNT    SCOPE:NAME                    STATE[OK/REPL/STUCK]    RSE_EXPRESSION           COPIES    SIZE      EXPIRES (UTC)        CREATED (UTC)
--------------------------------  ---------  ----------------------------  ----------------------  -----------------------  --------  --------  -------------------  -------------------
66d44ef919a84537a36ef1f0c5e9716d  test       test:1gbtestfile.20240219.bb  OK[1/0/0]               DUNE_US_FNAL_DISK_STAGE  1         1.024 GB  2024-02-22 03:07:21  2024-02-20 03:20:41
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio add-rule --account=test  test:1gbtestfile.20240219.bb  1 NIKHEF --lifetime 3600
cd5965017e564bf59621aa818fed72fa
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:test_expire_rules
ID    ACCOUNT    SCOPE:NAME    STATE[OK/REPL/STUCK]    RSE_EXPRESSION    COPIES    SIZE    EXPIRES (UTC)    CREATED (UTC)
----  ---------  ------------  ----------------------  ----------------  --------  ------  ---------------  ---------------
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio add-rule --account=test  test:test_expire_rules  1 NIKHEF --lifetime 3600
7e1abcd155ef4271b348e569cb3ecb82
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:test_expire_rules
ID                                ACCOUNT    SCOPE:NAME              STATE[OK/REPL/STUCK]    RSE_EXPRESSION    COPIES    SIZE    EXPIRES (UTC)        CREATED (UTC)
--------------------------------  ---------  ----------------------  ----------------------  ----------------  --------  ------  -------------------  -------------------
7e1abcd155ef4271b348e569cb3ecb82  test       test:test_expire_rules  OK[0/0/0]               NIKHEF            1         N/A     2024-02-20 04:21:57  2024-02-20 03:21:57
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio attach test:test_expire_rules test:1gbtestfile.20240219.bb
DIDs successfully attached to test:test_expire_rules
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:test_expire_rules
ID                                ACCOUNT    SCOPE:NAME              STATE[OK/REPL/STUCK]    RSE_EXPRESSION    COPIES    SIZE    EXPIRES (UTC)        CREATED (UTC)
--------------------------------  ---------  ----------------------  ----------------------  ----------------  --------  ------  -------------------  -------------------
7e1abcd155ef4271b348e569cb3ecb82  test       test:test_expire_rules  OK[0/0/0]               NIKHEF            1         N/A     2024-02-20 04:21:57  2024-02-20 03:21:57
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-files test:test_expire_rules
+------------------------------+--------------------------------------+-------------+------------+----------+
| SCOPE:NAME                   | GUID                                 | ADLER32     | FILESIZE   | EVENTS   |
|------------------------------+--------------------------------------+-------------+------------+----------|
| test:1gbtestfile.20240219.bb | 08F3F510-0BB8-4518-BB1F-EC6E13BFC73A | ad:93b40001 | 1.024 GB   |          |
+------------------------------+--------------------------------------+-------------+------------+----------+
Total files : 1
Total size : 1.024 GB
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio update-rule 7e1abcd155ef4271b348e569cb3ecb82 --lifetime 86400
Updated Rule
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-files test:test_expire_rules
+------------------------------+--------------------------------------+-------------+------------+----------+
| SCOPE:NAME                   | GUID                                 | ADLER32     | FILESIZE   | EVENTS   |
|------------------------------+--------------------------------------+-------------+------------+----------|
| test:1gbtestfile.20240219.bb | 08F3F510-0BB8-4518-BB1F-EC6E13BFC73A | ad:93b40001 | 1.024 GB   |          |
+------------------------------+--------------------------------------+-------------+------------+----------+
Total files : 1
Total size : 1.024 GB
[dunepro@dunesl7gpvm01 mcruciosam]$ rucio list-rules test:test_expire_rules
ID                                ACCOUNT    SCOPE:NAME              STATE[OK/REPL/STUCK]    RSE_EXPRESSION    COPIES    SIZE    EXPIRES (UTC)        CREATED (UTC)
--------------------------------  ---------  ----------------------  ----------------------  ----------------  --------  ------  -------------------  -------------------
7e1abcd155ef4271b348e569cb3ecb82  test       test:test_expire_rules  REPLICATING[0/1/0]      NIKHEF            1         N/A     2024-02-21 03:23:12  2024-02-20 03:21:5

**Comment by StevenCTimm** [comment]
The above reproducer didn't work but we were able to find one that did.

Have filed the issue in dune-justin GitHub https://github.com/DUNE/dune-justin/issues/7

basically it is the code on the Justin server side which is making the dataset and the associated rule at the beginning of the production and inadvertently adding an expired_at: on the dataset which leads to the dataset being deleted as well as everything in it as soon as that expired_at: field is met.  

Closing this, we'll track it in the Justin side.
